src/crimson/osd/pg_backend.cc

```diff
@@ -178,6 +178,70 @@ static inline bool _read_verify_data(
   return true;
 }
 
+repair_errorator::future<> PGBackend::rep_repair_primary_object(
+  const hobject_t& soid,
+  OSDOp& osd_op) 
+{
+  /*
+  TODO: 
+  1. Create and maintain the data structure `waiting_for_unreadable_object`
+  2. Get pool info
+  **/
+  const auto& op = osd_op.op
+  // Only supports replicated pools
+  ceph_assert(!pool.info.is_erasure());
+  ceph_assert(pg->is_primary());
+
+  logger().debug("{}: {} peers osd.{}", __func__, soid, get_acting_recovery_backfill());
+
+  if (!state_test(PG_STATE_CLEAN) && !pg->is_backfilling() && !pg->is_recovering()) {
+    logger().debug("{}: Blocked by PG state {}", __func__, soid);
+    block_for_clean(soid, op);
+    return crimson::ct_error::eagain::make();
+  }
+
+  ceph_assert(!pg->get_peering_state().get_pg_log().get_missing().is_missing(soid));
+  eversion_t v = ctx->obs->oi.version;
+
+  if (primary_error(soid, v)) {
+    logger().debug("{}: No other replicas available for {}", __func__, soid);
+    // XXX: If we knew that there is no down osd which could include this
+    // object, it would be nice if we could return EIO here.
+    // If a "never fail" flag was available, that could be used
+    // for rbd to NOT return EIO until object marked lost.
+
+    // Drop through to save this op in case an osd comes up with the object.
+  }
+
+  // Restart the op after object becomes readable again
+  waiting_for_unreadable_object[soid].push_back(op); // check if we have a similar mechanism in crimson potentially in client_request::process_op
```

Comment:
In classical-osd we have multiple queues that are responsible for preserving the order of blocked requests by requeueing them. `wait_for_unreadable_object` is one of them, containing all the requests for a missing object by mapping each object to its list of requests. Do we have a similar mechanism in Crimson? @tchaikov